### PR TITLE
[CARBONDATA-3873] Secondary index compaction with maintable clean files causing exception

### PR DESCRIPTION
**Pull request by maheshrajus**
[CARBONDATA-3873] Secondary index compaction with maintable clean files causing exception
 ### Why is this PR needed?
Compaction with secondary index and cleaning files of the main table after compaction fails then we are getting Exception.
If any compaction is failed for one of SI and if it is not disable then it will be considered for query execution. When query executed on SI table which is failed in compaction and it doesn't have correct data hence received Exception.
 
 ### What changes were proposed in this PR?
1) If any compaction is failed then disable all SI which are successful. They will be enabled after the next load.
2) Changing Error to the warning in table creation flow as it is a success case we should not give ERROR.

 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

    
https://issues.apache.org/jira/browse/CARBONDATA-3873

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1497/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3224/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3283/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1546/

File: core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java [hunk 1]
```diff
@@ -455,7 +455,7 @@ public boolean createNewLockFile() throws IOException {
     try {
       listStatus = fileSystem.listStatus(path);
     } catch (IOException e) {
-      LOGGER.error("Exception occured: " + e.getMessage(), e);
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java:458]: revert this change if not rerquired

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala [hunk 2]
```diff
@@ -121,10 +124,17 @@ object Compactor {
           segmentIdToLoadStartTimeMapping(validSegments.head),
           SegmentStatus.SUCCESS,
           carbonLoadModelForMergeDataFiles.getFactTimeStamp, rebuiltSegments.toList.asJava)
-
+        siCompactionIndexList ::= indexCarbonTable
       } catch {
         case ex: Exception =>
           LOGGER.error(s"Compaction failed for SI table ${secondaryIndex.indexName}", ex)
+          siCompactionIndexList.foreach { indexCarbonTable =>
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala:131]: please add a comment here, what you are doing in which scenario

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala [hunk 3]
```diff
@@ -121,10 +124,17 @@ object Compactor {
...
+            sparkSession.sql(
+              s"""ALTER TABLE ${carbonLoadModel.getDatabaseName}.${
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala:133]: please correct the format for sparksession.sql(""), please check the other files where we do alter set for SI, can follow the same

File: core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java (hunk 1 above)
Comment (VenuReddy2103) [reply core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java:458]: @akashrn5  Every time we create a new table/index/mv, we get this error log in the list files. This is the normal flow. Instead of error, warn is better to avoid confusion. I think, this looks ok. May be @maheshrajus can show this change details in the PR description.

Comment (maheshrajus) [reply core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java:458]: @akashrn5 @VenuReddy2103  i will add details in PR description

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala (hunk 2 above)
Comment (maheshrajus) [reply integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala:131]: OK

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala (hunk 3 above)
Comment (maheshrajus) [reply integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala:133]: OK

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala [hunk 4]
```diff
@@ -121,10 +124,20 @@ object Compactor {
...
-
+        siCompactionIndexList ::= indexCarbonTable
       } catch {
         case ex: Exception =>
           LOGGER.error(s"Compaction failed for SI table ${secondaryIndex.indexName}", ex)
+          // If any compaction is failed then make all SI disabled which are success.
+          // They will be enabled in next load
+          siCompactionIndexList.foreach { indexCarbonTable =>
+            sparkSession.sql(
+              s"""ALTER TABLE ${
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala:135]: format is not right, please correct it

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3309/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1571/

File: integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala [hunk 5]
```diff
@@ -121,10 +124,20 @@ object Compactor {
...
         case ex: Exception =>
           LOGGER.error(s"Compaction failed for SI table ${secondaryIndex.indexName}", ex)
+          // If any compaction is failed then make all SI disabled which are success.
+          // They will be enabled in next load
+          siCompactionIndexList.foreach { indexCarbonTable =>
+            sparkSession.sql(
+              s"""
+                 | ALTER TABLE ${carbonLoadModel.getDatabaseName}.${indexCarbonTable.getTableName}
+                 | SET
+                 | SERDEPROPERTIES ('isSITableEnabled' = 'false')
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala:138]: move this line above

**Comment by akashrn5** [comment]
LGTM

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3313/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1575/
